# SURYADIPTO SARKAR, MS | [Scholar](https://scholar.google.com/citations?user=I8mJjvsAAAAJ&hl=en&oi=ao) | [LinkedIn](https://www.linkedin.com/in/suryadipto-sarkar-ab1593197/?originalSubdomain=de)

PhD Candidate at FAU Erlangen, Bavaria, Germany. Due to submit his PhD dissertation by the end of Feb 2024.


# PERSONAL INFORMATION

## Address

Bayreutherstrasse 51, 91054 Erlangen, Bavaria, Germany ([directions](https://www.google.com/maps/place/Bayreuther+Str.+51,+91054+Erlangen/@49.6070816,10.9994195,17z/data=!3m1!4b1!4m6!3m5!1s0x47a1f93b60729999:0xf86688c63a61b700!8m2!3d49.6070816!4d11.0019944!16s%2Fg%2F11kpz_xbh7?entry=ttu&g_ep=EgoyMDI0MDkwOS4wIKXMDSoASAFQAw%3D%3D))

## Contact number

+49-15121585083

## Email ID

FAU: suryadipto.sarkar@fau.de

ASU: ssarka34@asu.edu

Personal: ssarkarmanipal@gmail.com


# ACADEMIC QUALIFICATION

 **PhD Candidate**, Friedrich-Alexander University of Erlangen-Nuremberg, Erlangen, Bavaria, Germany

 **Master of Science in Computer Engineering**, Arizona State University, Tempe, AZ, USA, May 2021, CGPA: 3.73/4

**Bachelor of Technology in Electronics and Communication Engineering**, Manipal Institute of Technology, Manipal, affiliated to Manipal University, Manipal, India, June 2016, CGPA: 7.99/10



# PROFESSIONAL EXPERIENCE

 **Junior Developer**, TEK Systems Global Services – Allegis Group, Bengaluru, India, August 2016 - January 2018

 **Trainee Engineer**, TEK Systems Global Services – Allegis Group, Bengaluru, India, January 2016 – May 2016


# ARTICLES UNDER REVIEW & ACCEPTED ARTICLES (first author)

(***Responding to round 1 revisions*** at [Nature NPJ Systems Biology and Applications](https://www.nature.com/npjsba/)) Sarkar, Suryadipto, et al. "Spatial cell graph analysis reveals skin tissue organization characteristic for cutaneous T cell lymphoma." bioRxiv (2024): 2024-05. ([preprint access here](https://www.biorxiv.org/content/10.1101/2024.05.17.594629v1.abstract))

(***Accepted with minor revisions*** at [MDPI Biomedicines](https://www.mdpi.com/journal/biomedicines), still in the process of being published) Sarkar, Suryadipto, et al. "A Transfer Learning-Based Framework for Classifying Lymph Node Metastasis in Prostate Cancer Patients." **(will be available online soon)**



# PUBLISHED ARTICLES (first author)

<ul>
	<li>Sarkar, Suryadipto, et al. "Performing automatic identification and staging of urothelial carcinoma in bladder cancer patients using a hybrid deep-machine learning approach." Cancers 15.6 (2023): 1673. ([access here](https://www.mdpi.com/2072-6694/15/6/1673))</li>
	<li>Sarkar, Suryadipto, et al. "Online bias-aware disease module mining with ROBUST-Web." Bioinformatics 39.6 (2023): btad345. ([access here](https://academic.oup.com/bioinformatics/article/39/6/btad345/7180269))</li>
	<li>Sarkar, Suryadipto, et al. "An advanced particle swarm optimization based feature selection method for tri-script handwritten digit recognition." Computational Intelligence, Communications, and Business Analytics: Second International Conference, CICBA 2018, Kalyani, India, July 27–28, 2018, Revised Selected Papers, Part I 2. Springer Singapore, 2019. ([access here](https://link.springer.com/chapter/10.1007/978-981-13-8578-0_7))</li>
	<li>[**Treatment response** – [Journal of Clinical Oncology](https://ascopubs.org/journal/jco)] Ikram, Waleed, et al. "Staging and chemotherapy response assessment of bladder cancer using hybrid deep machine learning model on CT scan images." (2022): e16550-e16550. ([access here](https://ascopubs.org/doi/abs/10.1200/JCO.2022.40.16_suppl.e16550))</li>
</ul>



# CO-AUTHORED PUBLISHED ARTICLES

<ul>
	<li>Fu, Yinlin, et al. "Gaussian mixture model with feature selection: An embedded approach." Computers & Industrial Engineering 152 (2021): 107000. ([access here](https://www.sciencedirect.com/science/article/pii/S0360835220306707?casa_token=XvzhD0Vmu6wAAAAA:uOLzxgEGcGEkp0ecwJl-ACs56MIoK_z8nzDkz8EOKscTZaAtGEO9g4NQ-wKCCpogjdFgu-52Tio))</li>
	<li>Xu, Yanzhe, et al. "U-net with optimal thresholding for small blob detection in medical images." 2019 IEEE 15th International Conference on Automation Science and Engineering (CASE). IEEE, 2019. ([access here](https://ieeexplore.ieee.org/abstract/document/8843234?casa_token=32Y0A1rGAL0AAAAA:5pdyOCRqW9yhL9SWUhz9wkSCMOs6uXWu3uDNSVom3VaDZjGfSMDdI2QahzDOSSAyx6aT2VXHyWw))</li>
	<li>Maier, Andreas, et al. "Drugst. One—a plug-and-play solution for online systems medicine and network-based drug repurposing." Nucleic Acids Research (2024): gkae388. ([access here](https://academic.oup.com/nar/article/52/W1/W481/7680613))</li>
</ul>



# CONFERENCES (first author)

## WITHOUT PROCEEDINGS

<ul>
	<li>[AIBE Conference 2022](https://www.aibe.tf.fau.de/2022/10/26/aibe-conf/) – presentation: (presentation not available anymore)</li>
	<li>GCB 2023 (Hamburg, Germany) – poster: ([access here](https://drive.google.com/file/d/1iFrQvKDbA8lAmzxGrtdqZxg4bZDB0UrS/view?usp=sharing))</li>
	<li>Same poster presented at the following conferences ([access here](https://drive.google.com/file/d/1GcD8lB3k3k6cpteqIisWvE30KNYwbKHZ/view?usp=sharing)):</li>
	<li>	
		<ul>
			<li>ECCB 2024 (Turku, Finland)</li>
			<li>GCB 2024 (Bielefeld, Germany)</li>
			<li>AIBE Conference (Erlangen, 2024)</li>
		</ul>
	</li>
	<li>[INFORMS Annual Meeting 2020](https://www.informs.org/Meetings-Conferences/INFORMS-Conference-Calendar/Virtual-2020-INFORMS-Annual-Meeting) – presentation: ([The Applications of Deep Learning in Medical Images for Efficient Diagnosis](https://auth.abstractsonline.com/SPLite/Account/ExternalLoginCallback?ReturnUrl=aHR0cHM6Ly93d3cuYWJzdHJhY3Rzb25saW5lLmNvbS9TUi9zcDJvZTIuYXNwP2FJZD00NzUmcmV0dXJuVXJsPWFIUjBjSE02THk5M2QzY3VZV0p6ZEhKaFkzUnpiMjVzYVc1bExtTnZiUzlrWVhOb1ltOWhjbVF2Ykc5bmFXNHVZWE53UDJGcFpEMDBOelVtZEdGeVoyVjBUVXRsZVQxN09EZERNakUwUWpRdFFqY3lPQzAwTmpnMUxVSkRNRGN0TlRNNE1UUkROemczTlRGR2ZTWjBZWEpuWlhSTmIyUTlKa0ZqWTNSTFpYazlKQ1JCWTJOMFMyVjVKQ1ElM2QmTT1EJlA9MSZNS2V5PSZUb2tlbj1PYXNpc1BlcnNvbiZJZHBFbnRpdHlJZD00NzU%3D&error=access_denied))</li>
	<li>[INFORMS Annual Meeting 2021](https://meetings.informs.org/wordpress/anaheim2021/) – presentation: (presentation not available anymore)</li>
</ul>


## WITH PROCEEDINGS

<ul>
	<li>Sarkar S., “Localization and Detection of Multi-lingual Text in Scenery Using an Ensemble of Gabor, FIR and Gaussian Filters,” International Journal of Scientific Research in Computer Science, Engineering and Information Technology, 2018, Volume 3, Issue 5, ISSN 2456-3307. ([access here](https://ijsrcseit.com/home/issue/view/article.php?id=CSEIT1835140))</li>
	<li>Sarkar S., Bhaumik S., Chatterjee S., “Identification of Malignant Tissues in Digital Mammogram Using an Automatically Generated Ensemble of Dynamic, Shifting Masks. ([article link](https://www.digitalxplore.org/up_proc/pdf/379-1531206177AllPapers.pdf)) ([conference proceedings link](https://digitalxplore.org/proceeding.php?pid=379))</li>
</ul>


# CONFERENCES (co-author)

## WITH PROCEEDINGS

<ul>
	<li>[ASCO 2022 Annual Meeting](https://meetings.asco.org/abstracts-presentations/search?q=*&filters=%7B%22meetingTypeName%22:%5B%7B%22key%22:%22ASCO%20Annual%20Meeting%22%7D%5D,%22meetingYear%22:%5B%7B%22key%22:%222022%22%7D%5D%7D): ([access published abstract here](https://meetings.asco.org/abstracts-presentations/209649))</li>
</ul>


## WITHOUT PROCEEDINGS

<ul>
	<li>[ARRS 2022](https://www.arrs.org/ARRSLIVE/StandAlone/ARRSLIVE/Education/2022_ARRS_Abstract_Submission.aspx): (abstract not available anymore)</li>
	<li>[Professor Baumbach](https://www.zbh.uni-hamburg.de/personen/csb/jbaumbach.html)’s group will be presenting the following papers at GCB 2024:
		<ul>[ROBUST-Web](https://academic.oup.com/bioinformatics/article/39/6/btad345/7180269) (presentation/ poster to be made available after conference)</ul>
	</li>
</ul>


# PREPRINTS

<ul>
	Identifying Malignant Lymph Nodes of Prostate Cancer Patients Using A Combination of Pre-Trained Deep Models and Traditional Machine Learning Classifiers ([access here](https://www.researchsquare.com/article/rs-143996/v1))
</ul>


# PAPERS UNDER DEVELOPMENT (first author, to be submitted soon)

<ul>
	<il>ENQUIRE-Web (target journal—[Bioinformatics](https://academic.oup.com/bioinformatics)):
		<ul><li>Status: Development done, code ready, manuscript pending.</li></ul>
	</il>
	<il>On mechanisms of gene co-expression (target journal—[PLOS Computational Biology](https://journals.plos.org/ploscompbiol/)):
		<ul>
			<li>Summary of the work: [access here](https://docs.google.com/document/d/1mcNjANS2JbYYHH6dlz8MPKTTJ--R91uIBbTfC3PngnU/edit)</li>
			<li>Main figure (Materials and Methods): [access here](https://drive.google.com/file/d/1T77p0zSQBQipAIfR_D7EiQHpBL4-7l15/view)</li>
			<li>Status: Development done, code ready, results semi-complete, manuscript pending.</li>
		</ul>
	</il>
</ul>



# ONLINE TOOLS

<ul>
	<il>**ROBUST-Web**: [https://robust-web.net/](https://robust-web.net/)</il>
	<il>**ENQUIRE-Web**: (will be made publicly available once paper has been submitted)</il>
	<li>**Drugst.One** (co-authored): [https://drugst.one/playground](https://drugst.one/playground)</li>
</ul>



PYTHON PACKAGES
•	SHouT (Spatial heterogeneity quantification tool): https://github.com/bionetslab/SHouT
o	Case studies:
1.	Cutaneous T-cell lymphoma (CTCL) case study: https://github.com/bionetslab/ctcl_case_study
•	ROBUST-bias-aware (the backend for ROBUST-Web): https://github.com/bionetslab/robust_bias_aware 
o	ROBUST-Web (python package for the front end that can alternatively be hosted locally from the command prompt/ terminal via localhost): https://github.com/bionetslab/robust-web  
o	Case studies already included in the app (see https://robust-web.net/run_robust): 
1.	COVID-19
2.	Precocious puberty
•	ENQUIRE-Web: https://github.com/bionetslab/enquire_web
•	GIST (Gene coexpression network insights)—will be made public once paper has been submitted: https://github.com/bionetslab/gcn-insights





<!------------------> 

# Install conda environment as follows (there also exists a requirements.txt)
```bash
conda create --name ctcl_case_study
conda activate ctcl_case_study
pip install scipy==1.10.1 numpy==1.23.5 squidpy==1.3.0 pandas==1.5.3 scikit-learn==1.2.2
```
*Note:* Additionally, modules *math* and *statistics* were used, however no installation is required as they are provided with Python by default.

<!------------------>

# HPA-based cell type assignment

## Steps involved:

**Algorithm:**

Step-1: Fit GMM model.

Step-2: Find list of good split genes.

Step-3: Calculate spread per celltype per gene in set {good\_split\_genes}.

Step-4: Arrange all genes in set {good\_split\_genes} in descending order, per celltype.

Step-5: Pick {gene-g, celltype-C} that maximizes spread.

Step-6: Assign g+: C.

Step-7: Repeat step-1.

**Once the algorithm is complete, do the following:**

i. Assign unassigned cells to 'Unknown' type.

ii. Map back assigned celltypes to the original cells.

iii. Save sample-wise cell type assignment results to a csv file and as an anndata.h5ad file.

*Note:* For a detailed explanation of the cell type assignment algorithm, please refer to the [paper](https://paper-doi-when-available).

## Running the code:

Navigate  to */scripts/hpa\_based\_cell\_type\_assignment/* and run **cell\_type\_assignment.py**.

## Output:

Sample-wise cell type assignment results saved as **/scripts/hpa\_based\_cell\_type\_assignment/result/sample-wise celltypes (HPA-based clustering).csv** and **/scripts/hpa\_based\_cell\_type\_assignment/result/celltype\_assigned\_anndata.h5ad**.

Additionally, if you want to save the results as separate .h5ad files per sample, please uncomment and run the last section of **cell\_type\_assignment.py** titled *Generate separate .h5ad files* (lines 77-86). This will result in separate .h5ad files saved as **/scripts/hpa\_based\_cell\_type\_assignment/result/<sample_id>.h5ad**. However, since these files have already been provided under the **/data** directory, upon which all of our subsequent SHouT heterogeneity score-based analyses have been performed, this code has been commented out by default in order to avoid generation of redundant .h5ad files.

*Note:* It must be noted that two of the samples, namely *291* and *294*, have been removed from the **/data** directory: that is simply because samples *291* and *294* contain too few segmented cells to run the SHouT heterogeneity scores on.

<!------------------>

# Generating and saving SHouT heterogeneity scores

## Computing the actual SHouT scores

Navigate  to */scripts/shout\_score\_generation/* and run **compute\_shout\_scores.py**.

New AnnData objects with SHouT scores saved in */results* folder as **.h5ad** files, with the same name as the original sample number.

## Saving SHouT scores as .csv files

Navigate  to */scripts/shout\_score\_generation/* and run **results\_to\_csv.py**.

Results are saved in */results/sample\_results.csv* for global (sample-wise) heterogeneity scores, and */results/cell\_results.csv* for local (individual cell-wise) heterogeneity scores.

## Statistical testing (Mann-Whitney U-test)

Navigate  to */scripts/shout\_score\_generation/* and run **statistical\_tests.py**.

Results are saved in */results/p\_values\_global.csv* for local (individual cell-wise) and global (sample-wise) heterogeneity scores with **r={1,2,3,4,5}** between all pairs of conditions; and */results/p\_values\_cell\_type.csv* for local (individual cell-wise) heterogeneity scores with **r={1,2,3,4,5}** and per celltype, between all pairs of conditions.

<!------------------>

# Robustness testing

## Shuffled labels

### Statistical testing (Mann-Whitney U-test)

Navigate  to */scripts/shout\_score\_generation/* and run **save\_pvalue\_statistics\_shuffled\_labels.py**.

Results are saved in */results/pvalue\_statistics\_shuffled\_labels.csv* for local (individual cell-wise) heterogeneity scores with **r={1,2,3,4,5}** between all pairs of conditions after having randomized the condition labels.


## Subsampled patients

### Statistical testing (Mann-Whitney U-test)

Navigate  to */scripts/shout\_score\_generation/* and run **statistical_tests_subsampled_patientwise.py**.

Results for local (individual cell-wise) heterogeneity scores are saved in */results/p_values_cell_type_subsampled_patientwise.csv* with **r={1,2,3,4,5}** between all pairs of conditions after having randomly subsampled 15 patients per condition at a time, while maintaining the actual condition labels, and repeating this subsampling process for 100 iterations.

Results for global (sample-wise) heterogeneity scores are saved in */results/p_values_global_subsampled_patientwise.csv* with **r={1,2,3,4,5}** between all pairs of conditions after having randomly subsampled 15 patients per condition at a time, while maintaining the actual condition labels, and repeating this subsampling process for 100 iterations.

<!------------------>

# Scalability testing

## Runtimes with varying radii

In order to record the runtimes of executing SHouT heterogeneity scores on all 69 samples, with radii r = {1, 5, 10, 20, 40, 80, 100}, navigate to */scripts/plots\_for\_paper/fig6\_scalability\_plot/* and run **per\_sample\_runtimes\_with\_varying\_radius.py**. The runtimes per sample, for each of r = {1, 5, 10, 20, 40, 80, 100}, is saved as */results/data\_all\_cells.csv*

<!------------------>

# Reproducing figures

## Reproducing results shown in Fig 2

Navigate  to */scripts/plots\_for\_paper/fig2\_database/* and run **fig2\_database.py**. Plot saved as *fig2\_database.pdf*.

## Reproducing results shown in Fig 3

Navigate  to */scripts/plots\_for\_paper/fig3\_subplot\_mosaic/* and run **fig3\_subplot\_mosaic.py**. Plot saved as *fig3\_subplot\_mosaic.pdf*.


## Reproducing results shown in Fig 4

Navigate  to */scripts/plots\_for\_paper/fig4\_shuffled\_labels/* and run **fig4\_shuffled\_labels.py**. Plot saved as *fig4\_shuffled\_labels.pdf*.

## Reproducing results shown in Fig 5

Navigate  to */scripts/plots\_for\_paper/fig5\_subsampled\_patients/* and run **fig5\_subsampled\_patients.py**. Plot saved as *fig5\_subsampled\_patients.pdf*.

## Reproducing results shown in Fig 6

Navigate  to */scripts/plots\_for\_paper/fig6\_scalability\_plot/* and run **fig6\_scalability\_plot.py**. Plot saved as *fig6\_scalability\_plot.pdf*.


## Reproducing results shown in Supplementary Fig 2

Navigate  to */scripts/plots\_for\_paper/supfig2\_cell\_abundance\_analyses/* and run **supfig2\_cell\_abundance\_analyses.py**. Plot saved as *supfig2\_cell\_abundance\_analyses.pdf*.

## Reproducing results shown in Supplementary Fig 3 - 5

Navigate  to */scripts/plots\_for\_paper/supfigs3,4,5\_SHouT\_scores\_radius\_5/* and run **generate\_supfig\_1\_2\_and\_3\_violinplots.py**. Plots saved as *SHouT\_score\_1\_violinplots.pdf*, *SHouT\_score\_2\_violinplots.pdf* and *SHouT\_score\_3\_violinplots.pdf* respectively.


## Reproducing results shown in Supplementary Fig 6

Navigate  to */scripts/plots\_for\_paper/supfig6\_SHouT\_scores\_all\_radii/* and run **supfig6\_SHouT\_scores\_all\_radii.py**. Plot saved as *supfig6\_SHouT\_scores\_all\_radii.pdf*.


## Reproducing results shown in Supplementary Fig 7

Navigate  to */scripts/plots\_for\_paper/supfig7\_centrality\_scores/* and run **generate\_supfig7\_centrality\_scores.py**. Plot saved as *supfig7\_centrality\_scores.pdf*.

<!------------------>

# Citing the work

Please cite our work as follows:
- Sarkar, S., Möller, A., Hartebrodt, A., Erdmann, M., Ostalecki, C., Baur, A. & Blumenthal, D. B. Spatial cell graph analysis reveals skin tissue organization characteristic for cutaneous T cell lymphoma. bioRxiv 2024.05.17.594629 (2024). doi:10.1101/2024.05.17.594629
